src/main/java/com/example/onlinebookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,67 @@
+package com.example.onlinebookstore.controller;
+
+import com.example.onlinebookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import com.example.onlinebookstore.service.BookService;
+import com.example.onlinebookstore.service.CategoryService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ADMIN')")
+    public CategoryDto save(@RequestBody CreateCategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @GetMapping
+    @PreAuthorize("hasRole('USER')")
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryService.findAll(pageable);
+    }
+
+    @GetMapping("/{id}")
+    @PreAuthorize("hasRole('USER')")
+    public CategoryDto findById(@PathVariable Long id) {
+        return categoryService.findById(id);
+    }
+
+    @PutMapping("/{id}")
+    @PreAuthorize("hasRole('ADMIN')")
+    public CategoryDto updateById(@PathVariable Long id,
+                                  @RequestBody CreateCategoryRequestDto requestDto) {
+        return categoryService.updateById(id, requestDto);
+    }

```

Comment:
Forgot to trigger validation.

```suggestion
    public CategoryDto updateById(@PathVariable Long id,
                                 @Valid @RequestBody CreateCategoryRequestDto requestDto) {
        return categoryService.updateById(id, requestDto);
    }

```